File: Backend/src/main/java/com/example/demo/Models/MessageModel.java
package com.example.demo.Models;


import java.util.Date;

public class MessageModel {

    private Long id;
    private String content;
    private Date dateOfSent;
    private ChatModel chatModel;
    private UserModel sender;

    public MessageModel() {
        this.dateOfSent = new Date();
    }

    public MessageModel(String content, Date dateOfSent) {
        this.content = content;
        this.dateOfSent = dateOfSent;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Date getDateOfSent() {
        return dateOfSent;
    }

    public void setDateOfSent(Date dateOfSent) {
        this.dateOfSent = dateOfSent;
    }

    public ChatModel getChatModel() {
        return chatModel;
    }

    public void setChatModel(ChatModel chatModel) {
        this.chatModel = chatModel;
    }

    public UserModel getSender() {
        return sender;
    }

    public void setSender(UserModel sender) {
        this.sender = sender;
    }

    @Override
    public String toString() {
        return "{\"id\":" + getId() + "" +
                ",\"content\":\"" + getContent() + "\"" +
                ",\"dateOfSent\":\"" + getDateOfSent() + "\"" +
                ",\"sender\":\"" + getSender() + "}";
    }
}

File: Frontend/lib/Model/MessageModel.dart
import 'ChatModel.dart';
import 'UserModel.dart';

class MessageModel {
  String? id;
  String? content;
  String? dateOfSent;
  ChatModel? chatModel;
  UserModel? sender;
  bool? unread;

  MessageModel({
    this.id,
    this.content,
    this.dateOfSent,
    this.chatModel,
    this.sender,
    this.unread,
  });

  MessageModel.fromJson(Map<String, dynamic> json) {
    id = json["id"].toString();
    content = json["content"];
    dateOfSent = json["dateOfSent"].toString();
    if (json["chatModel"] != null) {
      chatModel = ChatModel.fromJson(json["chatModel"]);
    }
    if (json["sender"] != null) {
      sender = UserModel.fromJson(json["sender"]);
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['content'] = content;
    data['dateOfSent'] = dateOfSent;
    if (chatModel != null) data['chatModel'] = chatModel!.toJson();
    if (sender != null) data['sender'] = sender!.toJson();
    return data;
  }

  get getId => id;

  set setId(String id) => this.id = id;

  get getContent => content;

  set setContent(content) => this.content = content;

  get getDateOfSent => dateOfSent;

  set setDateOfSent(dateOfSent) => this.dateOfSent = dateOfSent;

  get getChatModel => chatModel;

  set setChatModel(chatModel) => this.chatModel = chatModel;

  get getSender => sender;

  set setSender(sender) => this.sender = sender;

  get getUnread => unread;

  set setUnread(unread) => this.unread = unread;
}
